client/ayon_core/hosts/houdini/api/hda_utils.py

```diff
@@ -0,0 +1,402 @@
+    """Callback on thumbnail offset or size parms changed"""
+    thumbnail = lib.get_node_thumbnail(node)
+    if thumbnail:
+        rect = compute_thumbnail_rect(node)
+        thumbnail.setRect(rect)
+        lib.set_node_thumbnail(node, thumbnail)
+
+
+def on_representation_id_changed(node):
+    """Callback on representation id changed
+
+    Args:
+        node (hou.Node): Node to update.
+    """
+    repre_id = node.evalParm("representation")
+    set_representation(node, repre_id)
+
+
+def on_representation_parms_changed(node):
+    """
+    Usually used as callback to the project, folder, product, version and
+    representation parms which on change - would result in a different
+    representation id to be resolved.
+
+    Args:
+        node (hou.Node): Node to update.
+    """
+    project_name = node.evalParm("project_name") or get_current_project_name()
+    representation_id = get_representation_id(
+        project_name=project_name,
+        folder_path=node.evalParm("folder_path"),
+        product_name=node.evalParm("product_name"),
+        version=node.evalParm("version"),
+        representation_name=node.evalParm("representation_name"),
+        load_message_parm=node.parm("load_message")
+    )
+    if representation_id is None:
+        representation_id = ""
+    else:
+        representation_id = str(representation_id)
+
+    if node.evalParm("representation") != representation_id:
+        node.parm("representation").set(representation_id)
+        node.parm("representation").pressButton()  # trigger callback
+
+
+def get_representation_id(
+        project_name,
+        folder_path,
+        product_name,
+        version,
+        representation_name,
+        load_message_parm,
+):
+    """Get representation id.
+
+    Args:
+        project_name (str): Project name
+        folder_path (str): Folder name
+        product_name (str): Product name
+        version (str): Version name as string
+        representation_name (str): Representation name
+        load_message_parm (hou.Parm): A string message parm to report
+            any error messages to.
+
+    Returns:
+        Optional[str]: Representation id or None if not found.
+
+    """
+
+    if not all([
+        project_name, folder_path, product_name, version, representation_name
+    ]):
+        labels = {
+            "project": project_name,
+            "folder": folder_path,
+            "product": product_name,
+            "version": version,
+            "representation": representation_name
+        }
+        missing = ", ".join(key for key, value in labels.items() if not value)
+        load_message_parm.set(f"Load info incomplete. Found empty: {missing}")
+        return
+
+    try:
+        version = int(version.strip())
+    except ValueError:
+        load_message_parm.set(f"Invalid version format: '{version}'\n"
+                              "Make sure to set a valid version number.")
+        return
+
+    id_only = ["id"]
```

Comment:
Why to use variable?

Reply:
Just because it was re-used for each of the multiple queries. Fine with changing it around if you prefer that.

Reply:
Yes please.